src/neps/api.py

```diff
@@ -354,42 +334,46 @@ def _run_args(
                     if pipeline_space.has_fidelity
                     else "bayesian_optimization"
                 )
+            searcher_info[
+                "searcher_selection_source"
+            ] = "Default_Searcher-NePS_Decision_Tree"
         else:
             # Users choose one of NePS searchers.
-            user_defined_searcher = True
+            searcher_info["searcher_selection_source"] = "Default_Searcher-User_Choice"
         # Fetching the searcher data, throws an error when the searcher is not found
         config = get_searcher_data(searcher)
 
     searcher_alg = config["searcher_init"]["algorithm"]
-    searcher_config = {} if config["searcher_kwargs"] is None else config["searcher_kwargs"]
+    searcher_config = (
+        {} if config["searcher_kwargs"] is None else config["searcher_kwargs"]
+    )
 
     logger.info(f"Running {searcher} as the searcher")
     logger.info(f"Algorithm: {searcher_alg}")
 
     # Used to create the yaml holding information about the searcher.
     # Also important for testing and debugging the api.
-    searcher_info = {
-        "searcher_name": searcher,
-        "searcher_alg": searcher_alg,
-        "user_defined_searcher": user_defined_searcher,
-        "searcher_args_user_modified": False,
-    }
+    searcher_info["searcher_name"] = searcher
+    searcher_info["searcher_alg"] = searcher_alg
 
     # Updating searcher arguments from searcher_kwargs
     for key, value in searcher_kwargs.items():
-        if user_defined_searcher:
+        if (
+            searcher_info["searcher_selection_source"] == "Default_Searcher-User_Choice"
+            or searcher_info["searcher_selection_source"] == "Custom-User_Yaml"
+        ):
```

Comment:
These use of string matching will lead to a bug at somepoint, I would recommend some constants.